mm/2s2h/Enhancements/Cutscenes/StoryCutscenes/SkipLearningGoronLullaby.cpp

```diff
@@ -0,0 +1,97 @@
+#include <libultraship/bridge.h>
+#include "2s2h/GameInteractor/GameInteractor.h"
+#include "2s2h/CustomMessage/CustomMessage.h"
+#include "2s2h/CustomItem/CustomItem.h"
+#include "2s2h/Rando/Rando.h"
+#include "2s2h/ShipInit.hpp"
+
+extern "C" {
+#include "variables.h"
+#include "overlays/actors/ovl_En_Gk/z_en_gk.h"
+#include "overlays/actors/ovl_En_Go/z_en_go.h"
+#include "functions.h"
+
+void func_80B5227C(EnGk* enGkActor, PlayState* play);
+void EnGo_Sleep(EnGo* enGoActor, PlayState* play);
+}
+
+#define CVAR_NAME "gEnhancements.Cutscenes.SkipStoryCutscenes"
+#define CVAR CVarGetInteger(CVAR_NAME, 0)
+
+static bool isGoronSleepQueued = false;
+
+// This is a song tutorial, so the skip is forced on in rando for now
+void RegisterSkipLearningGoronLullaby() {
+    // Played Lullaby Intro for Baby Goron
+    COND_VB_SHOULD(VB_START_CUTSCENE, CVAR || IS_RANDO, {
+        s16* csId = va_arg(args, s16*);
+        Actor* actor = va_arg(args, Actor*);
+
+        if (*csId != 9 || actor == NULL || actor->id != ACTOR_EN_GK) {
+            return;
+        }
+
+        EnGk* enGk = (EnGk*)actor;
+
+        // Reset baby Goron cutscene flags state
+        enGk->unk_1E4 &= ~0x20;
+        *should = false;
+
+        // Set action func which puts Baby to sleep, sets calm week event flag
+        enGk->csAnimIndex = 1; // ENGK_ANIM_1
+        enGk->animIndex = 1;   // ENGK_ANIM_1
+        enGk->actionFunc = func_80B5227C;
+
+        // Activate torches
+        Flags_SetSwitch(gPlayState, ENGK_GET_SWITCH_FLAG(&enGk->actor));
+
+        isGoronSleepQueued = true;
+
+        if (IS_RANDO) {
+            SET_WEEKEVENTREG(WEEKEVENTREG_24_80); // Ensure Goron Elder check is available
+            RANDO_SAVE_CHECKS[RC_GORON_SHRINE_FULL_LULLABY].eligible = true;
+        } else {
```

Comment:
Similar here to the other PR, rando related logic should not be happening within these files